### PR TITLE
Add option to prevent node from outputting deprecation warnings

### PR DESCRIPTION
**Pull request by velrest**
Since the output of eslint should not contain anything more than linting errors, we should be able to pass node custom options to f.e. disable deprecation warnings.

Solution for https://github.com/dense-analysis/ale/issues/2891

### CODE REVIEW

**Comment by elebow** [comment]
I think we always want to prevent warning output from nodejs, for all linters. We could probably do something like this:

```diff
diff --git a/autoload/ale/node.vim b/autoload/ale/node.vim
index 69060122..929be499 100644
--- a/autoload/ale/node.vim
+++ b/autoload/ale/node.vim
@@ -35,7 +35,9 @@ function! ale#node#Executable(buffer, executable) abort
     if has('win32') && a:executable =~? '\.js$'
         let l:node = ale#Var(a:buffer, 'windows_node_executable_path')
 
-        return ale#Escape(l:node) . ' ' . ale#Escape(a:executable)
+        return ale#Escape(l:node)
+        \   . ' --no-deprecation '
+        \   . ale#Escape(a:executable)
     endif
 
     return ale#Escape(a:executable)
```

But that's a partial solution, only for certain situations on Windows. Most of the time, node binaries are not invoked through `node`, but rather through an executable script with a `#!/usr/bin/node` hashbang.

In that case, there is no opportunity to pass command-line arguments to the node interpreter, but we may be able to use the `NODE_NO_WARNINGS` environment variable. I'm not sure exactly where this should be set in ALE, or even if it's a good approach.

An example, just to illustrate the concept—this is obviously not correct:

```diff
diff --git a/autoload/ale/engine.vim b/autoload/ale/engine.vim
index 491d3c2e..be33af53 100644
--- a/autoload/ale/engine.vim
+++ b/autoload/ale/engine.vim
@@ -450,6 +450,8 @@ function! s:RunJob(command, options) abort
         return 0
     endif
 
+    let $NODE_NO_WARNINGS = 1
+
     let l:executable = a:options.executable
     let l:buffer = a:options.buffer
     let l:linter = a:options.linter
```

**Comment by elebow** [comment]
Or actually, we _could_ invoke `node` every time, like this PR.

This might be all we need:

```diff
diff --git a/autoload/ale/node.vim b/autoload/ale/node.vim
index 69060122..997a7e51 100644
--- a/autoload/ale/node.vim
+++ b/autoload/ale/node.vim
@@ -34,9 +34,11 @@ endfunction
 function! ale#node#Executable(buffer, executable) abort
     if has('win32') && a:executable =~? '\.js$'
         let l:node = ale#Var(a:buffer, 'windows_node_executable_path')
-
-        return ale#Escape(l:node) . ' ' . ale#Escape(a:executable)
+    else
+        let l:node = 'node'
     endif
 
-    return ale#Escape(a:executable)
+    return ale#Escape(l:node)
+    \   . ' --no-deprecation '
+    \   . ale#Escape(a:executable)
 endfunction
```

There's probably an edge case there on non-Unixy platforms, but I think that general approach would work fine.

**Comment by velrest** [comment]
Is there a better way to fix the test than prefixing every Assert that executes a node command with `node --no-deprecation`?

**Comment by elebow** [comment]
It looks like https://github.com/dense-analysis/ale/pull/2910 would be a more reliable solution to the problem, since it works even if users override the eslint executable.

If you were going to fix the tests in this PR, then yes you would have to mass replace the asserted values.

**Comment by velrest** [comment]
Okay thanks. Closing since there is already a solution.
